itest/mint_fund_seal_test.go

```diff
@@ -0,0 +1,706 @@
+	// input, which is the genesis output from the minting batch. We'll
+	// satisfy the 2-of-2 multisig script we specified when funding the
+	// batch.
+	// To build the control block, we need the internal key from that
+	// output, and the Taproot Asset root hash used for that batch. This
+	// output is currently leased for the transfer we're constructing.
+	assetInputIdx := uint32(0)
+	aliceUtxoResp, err := aliceTapd.ListUtxos(
+		ctxt, &taprpc.ListUtxosRequest{
+			IncludeLeased: true,
+		},
+	)
+	require.NoError(t.t, err)
+	require.Len(t.t, aliceUtxoResp.ManagedUtxos, 1)
+	anchorOutpoint := wire.OutPoint{
+		Hash:  batchTXID,
+		Index: 0,
+	}
+	anchorUTXO, ok := aliceUtxoResp.ManagedUtxos[anchorOutpoint.String()]
+	require.True(t.t, ok)
+
+	anchorInputInternalKey, err := btcec.ParsePubKey(anchorUTXO.InternalKey)
+	require.NoError(t.t, err)
+
+	anchorInputOutputKey := txscript.ComputeTaprootOutputKey(
+		anchorInputInternalKey, anchorUTXO.MerkleRoot,
+	)
+	anchorKeyYIsOdd := anchorInputOutputKey.SerializeCompressed()[0] ==
+		secp256k1.PubKeyFormatCompressedOdd
+
+	anchorInputControlBlock := txscript.ControlBlock{
+		InternalKey:     anchorInputInternalKey,
+		OutputKeyYIsOdd: anchorKeyYIsOdd,
+		LeafVersion:     txscript.BaseLeafVersion,
+		InclusionProof:  anchorUTXO.TaprootAssetRoot,
+	}
+	anchorInputControlBlockBytes, err := anchorInputControlBlock.ToBytes()
+	require.NoError(t.t, err)
+
+	// With our control block, we can generate the two needed signatures
+	// and construct the anchor witness.
+	firstKeyPartialSig := partialSignWithKey(
+		t.t, aliceLnd.RPC, regtestParams, transferPkt, assetInputIdx,
+		firstAnchorInternal, anchorInputControlBlockBytes,
+		anchorMultisigLeaf,
+	)
+	secondKeyPartialSig := partialSignWithKey(
+		t.t, aliceLnd.RPC, regtestParams, transferPkt, assetInputIdx,
+		secondAnchorInternal, anchorInputControlBlockBytes,
+		anchorMultisigLeaf,
+	)
+
+	anchorInputWitness := wire.TxWitness{
+		secondKeyPartialSig,
+		firstKeyPartialSig,
+		anchorMultisigScript,
+		anchorInputControlBlockBytes,
+	}
+	var witBuf bytes.Buffer
+	err = psbt.WriteTxWitness(&witBuf, anchorInputWitness)
+	require.NoError(t.t, err)
+
+	transferPkt.Inputs[assetInputIdx].FinalScriptWitness = witBuf.Bytes()
+
+	// Finalize and publish the transfer anchor TX.
+	signedPkt := FinalizePacket(t.t, aliceLnd.RPC, transferPkt)
+	logResp := LogAndPublish(
+		t.t, aliceTapd, signedPkt, bobVpsbt, passiveVpsbts,
+		commitResp,
+	)
+	t.Logf("Logged transaction: %v", toJSON(t.t, logResp))
+
+	// Mine a block to confirm the transfer.
+	MineBlocks(t.t, t.lndHarness.Miner.Client, 1, 1)
+
+	// Bob should have detected the transfer, and now own half of the total
+	// asset supply.
+	AssertAddrEvent(t.t, bobTapd, bobAddr, 1, statusCompleted)
+	AssertNonInteractiveRecvComplete(t.t, bobTapd, 1)
+	AssertBalanceByID(
+		t.t, bobTapd, assetTweakedScriptKey.AssetGenesis.AssetId,
+		assetTweakedScriptKey.Amount/2,
+	)
+
+	time.Sleep(time.Second / 4)
```

Comment:
Can we use `ConfirmAndAssertOutboundTransferWithOutputs` instead of the sleep to make sure Alice sees the outbound transfer as complete as well?